core/controllers/dashboard_test.py

```diff
@@ -66,6 +75,349 @@ def test_logged_in_notifications_dashboard(self):
+        self._rate_exploration(exp_id_1, 1, rating_start)
+
+        self._start_computation_job()
+        user_model = user_models.UserStatsModel.get(
+            self.owner_id_1, strict=False)
+        self.assertEquals(user_model.total_plays, 1)
+        self.assertEquals(
+            user_model.impact_score, self._get_default_value_of_impact_score())
+        self.assertEquals(user_model.average_ratings, 4)
+        self.logout()
+
+    def test_multiple_plays_and_ratings_for_multiple_explorations(self):
+        exploration_1 = self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id_1, title=self.EXP_TITLE_1)
+
+        exploration_2 = self.save_new_default_exploration(
+            self.EXP_ID_2, self.owner_id_1, title=self.EXP_TITLE_2)
+
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 2)
+
+        exp_version = self._get_default_version_of_exploration()
+
+        exp_id_1 = self.EXP_ID_1
+        state_1 = exploration_1.init_state_name
+        exp_id_2 = self.EXP_ID_2
+        state_2 = exploration_2.init_state_name
+
+        self._record_start(exp_id_1, exp_version, state_1)
+        self._record_start(exp_id_2, exp_version, state_2)
+        self._record_start(exp_id_2, exp_version, state_2)
+        self._start_job_to_compute_num_plays()
+
+        rating_start_1 = 4
+        self._rate_exploration(exp_id_1, 1, rating_start_1)
+        rating_start_2 = 3
+        self._rate_exploration(exp_id_2, 1, rating_start_2)
+
+        self._start_computation_job()
+        user_model = user_models.UserStatsModel.get(
+            self.owner_id_1, strict=False)
+        self.assertEquals(user_model.total_plays, 3)
+        self.assertEquals(
+            user_model.impact_score, self._get_default_value_of_impact_score())
+        self.assertEquals(user_model.average_ratings, 3.5)
+        self.logout()
+
+    def test_stats_for_single_exploration_with_multiple_owners(self):
+        exploration = self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id_1, title=self.EXP_TITLE_1)
+
+        rights_manager.assign_role_for_exploration(
+            self.owner_id_1, self.EXP_ID_1, self.owner_id_2,
+            rights_manager.ROLE_OWNER)
+
+        self.login(self.OWNER_EMAIL_2)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(len(response['explorations_list']), 1)
+
+        exp_version = self._get_default_version_of_exploration()
+        exp_id = self.EXP_ID_1
+        state = exploration.init_state_name
+
+        self._record_start(exp_id, exp_version, state)
+        self._record_start(exp_id, exp_version, state)
+        self._start_job_to_compute_num_plays()
+
+        rating_start = 3
+        self._rate_exploration(exp_id, 3, rating_start)
+
+        self._start_computation_job()
+        user_model = user_models.UserStatsModel.get(
```

Comment:
I don't think that checking for user 2 will be required here, since user 2 is nowhere involved in this particular test. What do you think ?